pyvisa/testsuite/test_util.py

```diff
@@ -509,80 +510,84 @@ def get_debug_info(cls):
         util.system_details_to_str(details)
 
 
-def generate_fakelibs(dirname):
-    for name, blob in zip(
-        [
-            "fakelib_bad_magic.dll",
-            "fakelib_good_32.dll",
-            "fakelib_good_64_2.dll",
-            "fakelib_good_64.dll",
-            "fakelib_good_unknown.dll",
-            "fakelib_not_pe.dll",
-        ],
-        [
-            struct.pack("=6sH52sl", b"MAPE\x00\x00", 0x014C, 52 * b"\0", 2),
-            struct.pack("=6sH52sl", b"MZPE\x00\x00", 0x014C, 52 * b"\0", 2),
-            struct.pack("=6sH52sl", b"MZPE\x00\x00", 0x8664, 52 * b"\0", 2),
-            struct.pack("=6sH52sl", b"MZPE\x00\x00", 0x0200, 52 * b"\0", 2),
-            struct.pack("=6sH52sl", b"MZPE\x00\x00", 0xFFFF, 52 * b"\0", 2),
-            struct.pack("=6sH52sl", b"MZDE\x00\x00", 0x014C, 52 * b"\0", 2),
-        ],
-    ):
-        with open(os.path.join(dirname, name), "wb") as f:
+def generate_fakelibs(dirname: Path):
+    libs = {
+        "fakelib_bad_magic.dll": struct.pack(
+            "=6sH52sl", b"MAPE\x00\x00", 0x014C, 52 * b"\0", 2
+        ),
+        "fakelib_good_x86_32.dll": struct.pack(
+            "=6sH52sl", b"MZPE\x00\x00", 0x014C, 52 * b"\0", 2
+        ),
+        "fakelib_good_x86_64.dll": struct.pack(
+            "=6sH52sl", b"MZPE\x00\x00", 0x8664, 52 * b"\0", 2
+        ),
+        "fakelib_good_arm_64.dll": struct.pack(
+            "=6sH52sl", b"MZPE\x00\x00", 0xAA64, 52 * b"\0", 2
+        ),
+        "fakelib_good_unknown.dll": struct.pack(
+            "=6sH52sl", b"MZPE\x00\x00", 0xFFFF, 52 * b"\0", 2
+        ),
+        "fakelib_not_pe.dll": struct.pack(
+            "=6sH52sl", b"MZDE\x00\x00", 0x014C, 52 * b"\0", 2
+        ),
+    }
+    for name, blob in libs.items():
+        with open(Path(dirname, name), "wb") as f:
             f.write(blob)
             print("Written %s" % name)
 
 
 class TestLibraryAnalysis(BaseTestCase):
     """Test (through monkey patching) the analysis of binary libraries."""
 
-    def test_get_shared_library_arch(self, tmpdir):
+    def test_get_shared_library_arch(self, tmp_path: Path):
         """Test analysing a library on Windows."""
-        dirname = str(tmpdir)
-        generate_fakelibs(dirname)
-
-        for f, a in zip(["_32", "_64", "_64_2"], ["I386", "IA64", "AMD64"]):
+        generate_fakelibs(tmp_path)
+
+        for f, a in zip(
+            ["x86_32", "x86_64", "arm_64"],
+            [
+                util.PEMachineType.I386,
+                util.PEMachineType.AMD64,
+                util.PEMachineType.AARCH64,
+            ],
+        ):
             arch = util.get_shared_library_arch(
-                os.path.join(tmpdir, "fakelib_good%s.dll" % f)
+                Path(tmp_path, "fakelib_good_%s.dll" % f)
```

Comment:
Since tmp_path is a Path, simply use the / based syntax.

Reply:
oh, good call